pkg/virt-handler/vm.go

```diff
@@ -439,7 +440,51 @@ func MapPersistentVolumes(vm *v1.VirtualMachine, clientset kubecli.KubevirtClien
 }
 
 func mapPVToDisk(disk *v1.Disk, pv *k8sv1.PersistentVolume) (*v1.Disk, error) {
-	if pv.Spec.ISCSI != nil {
+	if pv.Spec.Glusterfs != nil {
+		newDisk := v1.Disk{}
+		newDisk.Type = "network"
+		newDisk.Device = "disk"
+		newDisk.Target = disk.Target
+		newDisk.Driver = new(v1.DiskDriver)
+		newDisk.Driver.Type = "raw"
+		newDisk.Driver.Name = "qemu"
+
+		//Set Gluster Connection and libgfapi URI.
+		// VM image file name should be formatted or fetched to accomodate the vmspec name or UUID.
+		glusterImagePath := fmt.Sprintf("%s/vm-%d", pv.Spec.Glusterfs.Path, uuid.NewUUID())
+		ipAddress := ""
+		addrHosts := pv.Spec.PersistentVolumeSource.Glusterfs.endpoints
+		var addrlist []string
+		if addrHosts == nil {
+			return nil, fmt.Errorf("glusterfs: endpoint is nil")
+		}
+		addr := sets.String{}
+		if addrHosts.Subsets != nil {
+			for _, s := range addrHosts.Subsets {
+				for _, a := range s.Addresses {
+					if !addr.Has(a.IP) {
+						addr.Insert(a.IP)
+						addrlist = append(addrlist, a.IP)
+					}
+				}
+			}
+		}
+		if (len(addrlist) > 0) && (addrlist[0] != "") {
+			ipAddress = addrlist[0]
+		}
+		glusterURI := fmt.Sprintf("%s://%s:24007/%s", "gluster", ipAddress, glusterImagePath)
+		//TODO: Check if the file is present.
+		cmd := exec.Command("qemu-img", "create", glusterURI, "5G")
```

Comment:
We'd expect the gluster storage to already be created and populated with the disk image contents before the VM is scheduled to start.

This function just maps the technology specific bits that back a PV into something libvirt can consume.  It shouldn't be allocating anything on block storage.

Reply:
@davidvossel Thanks. `We'd expect the gluster storage to already be created and populated with the disk image contents before the VM is scheduled to start.` -> Iiic, in this case, we need to pass `vmspec` to mapPVToDisk(). If not, is there a way to find out the vm disk image name in mapPVToDisk()?

Reply:
@humblec  Ha, well there isn't a disk image name in the VM api right now really. You're working in some area's that haven't had their design finalized yet (both in the API and backend).

This mapPVToDisk function is proof of concept code. This method isn't necessarily reflective of the long term strategy.